environs/bootstrap/bootstrap.go

```diff
@@ -659,15 +659,33 @@ func isCompatibleVersion(v1, v2 version.Number) bool {
 // for tools syncing, and adds an image metadata source after verifying
 // the contents.
 func setPrivateMetadataSources(metadataDir string) ([]*imagemetadata.ImageMetadata, error) {
-	logger.Infof("Setting default tools and image metadata sources: %s", metadataDir)
-	tools.DefaultBaseURL = metadataDir
+	if _, err := os.Stat(metadataDir); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Annotate(err, "cannot access metadata directory")
+		}
+		logger.Infof("metadata source: %s, not found", metadataDir)
```

Comment:
We should return a not found error here so it bubbles up and aborts bootstrap
return errors.NotFoundf("simplestreams metadata source directory %v", metadataDir)